cli/cmd/deploy/deploy.go

```diff
@@ -183,11 +197,27 @@ func DeployCmd(cfg *config.Config) *cobra.Command {
 	deployCmd.Flags().StringVar(&region, "region", "", "Deployment region")
 	deployCmd.Flags().StringVar(&dbDriver, "prod-db-driver", "duckdb", "Database driver")
 	deployCmd.Flags().StringVar(&dbDSN, "prod-db-dsn", "", "Database driver configuration")
+	deployCmd.Flags().BoolVar(&public, "public", false, "Make dashboards publicly accessible")
+	deployCmd.Flags().StringVar(&prodBranch, "prod-branch", "", "Git branch to deploy from (default: the default Git branch)")
+	deployCmd.Flags().StringVar(&name, "name", "", "Project name (default: taken from rill.yaml)")
 	return deployCmd
 }
 
-func projectParamPrompt(ctx context.Context, c *client.Client, orgName, githubURL, prodBranch string) (*promptOptions, error) {
-	projectName := path.Base(githubURL)
+func projectNamePrompt(ctx context.Context, c *client.Client, orgName, projectPath string) (string, error) {
+	projectName, err := projectName(projectPath)
+	if err != nil {
+		return "", err
+	}
+
+	exist, err := projectExists(ctx, c, orgName, projectName)
+	if err != nil {
+		return "", err
+	}
+
+	if !exist {
+		fmt.Printf("Creating project by name %q. Use `rill project edit` to edit name if required.\n", projectName)
+		return projectName, nil
+	}
```

Comment:
1. Its only printed in cases when we auto-chose the name sort of like FYI. We can remove this log.
2. The problem with doing this would be that we would consider all kind of failures as project already exists by name which may be okay but not ideal. May be lets refactor it once we refactor admin backend to give better error messages and we can differentiate between duplicate name vs other errors?

Reply:
1. I like the log, I just think if I see a "Creating..." log, I expect it to be followed by a "Created..." log upon success. Since the action runs so quickly, having two feels like overkill – so I think it makes more sense to have "Created..." since then you know it succeeded.
2. Unique constraint violations will always return `database.ErrNotUnique` as an error. So you can check for that here.

Reply:
1. Yeah makes sense. I removed the earlier log now. 
2. We seem to be overriding the error with `codes.InvalidArgument` here : https://github.com/rilldata/rill-developer/blob/933c7bcca003061cff93e3614a939bd702b33d60/admin/server/projects.go#L154